vignettes/col_counts.Rmd

```diff
@@ -0,0 +1,130 @@
+  analyze("AGE", afun = max, format = "xx.x")
+
+tbl <- build_table(lyt, example_data)
+tbl
+```
+
+We can control the column counts with the following functions:
+
+```{r}
+col_counts(tbl) # vector of all of them (visible and not visible) -> difficult to read/use for nested structures
+coldf <- make_col_df(tbl) # to check the structure of the columns as a data.frame
+coltree_structure(tbl) # Prints the structure of the columns as a tree
+```
+
+`make_col_df` is a fundamental function that is used also internally to keep track of the tree structure of the columns. It returns a `data.frame` very similar to `make_row_df`, but for columns. The `coltree_structure` function is a wrapper around `make_col_df` that prints the structure of the columns as a tree.
+
+To retrieve the column counts for a specific column, use pathing! Also visibility can be accessed and modified with the following functions:
+
+```{r}
+facet_colcount(tbl, coldf$path[[1]]) # To get the column counts for a specific column at a specific level
+colcount_visible(tbl, coldf$path[[1]]) # not visible!!! # FALSE
+facet_colcount(tbl, coldf$path[[1]][c(1, 2)]) # visible
+colcount_visible(tbl, coldf$path[[1]][c(1, 2)]) # TRUE
+```
+
+Formats can differ as we have shown in the example above. We can change the format of the column counts with the following function:
+
+```{r, eval = FALSE}
+# Modifying last level!?
+colcount_format(tbl) <- "[N<=xx]" # Does not work
+colcount_visible(tbl, coldf$path[[1]]) <- TRUE # not the right way!!
+# > tbl # ERROR! we need to set the other values too
+# Error in h(simpleError(msg, call)) :
+#   error in evaluating the argument 'x' in selecting a method for function 'toString':
+#   Detected different colcount visibility among sibling facets (those arising from the
+#   same split_cols_by* layout instruction). This is not supported.
+# Set count values to NA if you want a blank space to appear as the displayed count for particular facets.
+# First disagreement occured at paths:
+# ARMCD[ARM A]->STRATA2[S1]->STRATA1[A]
+# ARMCD[ARM A]->STRATA2[S1]->STRATA1[B]
+# colcount_visible(tbl, coldf$path[[1]]) <- FALSE
+
+# Trying with col_counts
+col_counts(tbl) <- c(1, rep(NA_integer_, length(col_counts(tbl)) - 1)
+# colcount_visible(tbl, coldf$path[[1]]) <- TRUE # same error as above we need facets!!
+facet_colcounts_visible(tbl, c("ARMCD", "*", "STRATA2", "*", "STRATA1")) <- TRUE # does not work
+print(tbl)
+```
+
+It is also possible to set the column counts and visibility for a specific column level with pathing and the following function:
+
+```{r}
+# Modifying another level/column
+facet_colcount(tbl, c("ARMCD", "ARM A")) <- NA_integer_ # To remove the column counts for a specific column
+# facet_colcount(tbl, "ARMCD") # not working
+print(tbl)
+
+# colcount_visible(tbl, c("*")) <- TRUE # it does not work!! We need facet_colcount_visible
+facet_colcounts_visible(tbl, c("ARMCD")) <- FALSE  # function needs examples and connection to colcounts_visible
```

Comment:
`facet_colcounts_visible` needs more in-function documentation with examples and connections to its counterparts (`colcounts_visible` and `facet_colcounts` and `col_counts`)

Reply:
done